### PR TITLE
[CustomDevice] add bf16 ccltype for custom device

### PR DESCRIPTION
**Pull request by KimBioInfoStudio**
### PR types

Others
### PR changes

Others
### Description

add bf16 ccltype for custom device

### CODE REVIEW

**Comment by KimYannn** [comment]
cc @jinyouzhi @kevinyuk

File: paddle/phi/backends/c_comm_lib.h [hunk 1]
```diff
@@ -73,9 +77,40 @@ inline phi::DataType ToPhiDataType(CCLDataType type) {
     return phi::DataType::INT32;
   } else if (type == CCLDataType::CCL_DATA_TYPE_INT8) {
```
Comment (jinyouzhi) [inline paddle/phi/backends/c_comm_lib.h:107]: missing INT16

File: paddle/phi/backends/c_comm_lib.h [hunk 2]
```diff
@@ -73,9 +77,40 @@ inline phi::DataType ToPhiDataType(CCLDataType type) {
...
+      return "float64";
+    case CCLDataType::CCL_DATA_TYPE_FP32:
+      return "float32";
+    case CCLDataType::CCL_DATA_TYPE_FP16:
+      return "float16";
+    case CCLDataType::CCL_DATA_TYPE_INT64:
+      return "int64";
+    case CCLDataType::CCL_DATA_TYPE_INT32:
+      return "int32";
+    case CCLDataType::CCL_DATA_TYPE_INT8:
```
Comment (jinyouzhi) [inline paddle/phi/backends/c_comm_lib.h]: missing INT16

File: paddle/phi/backends/c_comm_lib.h [hunk 3]
```diff
@@ -53,6 +55,8 @@ inline CCLDataType ToCCLDataType(phi::DataType type) {
     return CCL_DATA_TYPE_INT8;
   } else if (type == phi::DataType::UINT8) {
```
Comment (jinyouzhi) [inline paddle/phi/backends/c_comm_lib.h:57]: missing INT16

**Review by YanhuiDua — APPROVED** [review]
LGTM

**Comment by KimYannn** [comment]
@luotao1 pls help review

**Pull request closed without merging**